libraries/classes/Html/Generator.php

```diff
@@ -1369,26 +1372,32 @@ public static function getSupportedDatatypes($selected): string
                         $retval .= '</option>';
                     } else {
                         $retval .= sprintf(
-                            '<option title="%s">%s</option>',
+                            '<option data-length-restricted="%b" title="%s">%s</option>',
+                            $isLengthRestricted,
                             $dbi->types->getTypeDescription($subvalue),
                             $subvalue
                         );
                     }
                 }
 
                 $retval .= '</optgroup>';
-            } elseif ($selected == $value) {
-                $retval .= sprintf(
-                    '<option selected="selected" title="%s">%s</option>',
-                    $dbi->types->getTypeDescription($value),
-                    $value
-                );
             } else {
-                $retval .= sprintf(
-                    '<option title="%s">%s</option>',
-                    $dbi->types->getTypeDescription($value),
-                    $value
-                );
+                $isLengthRestricted = Compatibility::isIntegersSupportLength($value, '2', $dbi) === true ? 0 : 1;
+                if ($selected == $value) {
+                    $retval .= sprintf(
+                        '<option data-length-restricted="%b" selected="selected" title="%s">%s</option>',
+                        $isLengthRestricted,
+                        $dbi->types->getTypeDescription($value),
+                        $value
+                    );
+                } else {
+                    $retval .= sprintf(
+                        '<option data-length-restricted="%b" title="%s">%s</option>',
+                        $isLengthRestricted,
+                        $dbi->types->getTypeDescription($value),
+                        $value
+                    );
+                }
```

Comment:
I think this if else could be removed by using this
```suggestion
                $retval .= sprintf(
                    '<option data-length-restricted="%b" %s title="%s">%s</option>',
                    $isLengthRestricted,
                    $selected === $value ? 'selected="selected"' : ''
                    $dbi->types->getTypeDescription($value),
                    $value
                );
```

Reply:
Looks much better, thank you.